cognito-idp client supports filter expressions unsupported by real boto client

I wanted to use `list_users` function from `cognito-idp` client API in my code in the following way:
```
boto3.client('cognito-idp', aws_access_key_id='redacted', aws_secret_access_key='redacted').list_users(UserPoolId=os.getenv('COGNITO_USER_POOL_ID'), Filter='preferred_username=nickname0')
```

So I started by writing unit test for the function with moto mocks applied. Fixtures were used in the test:
```
@pytest.fixture(scope='function')
def cognito_idp(aws_credentials):
    with mock_cognitoidp():
        yield boto3.client('cognito-idp', region_name='us-east-1')


@pytest.fixture(scope='function')
def cognito_user_pool(cognito_idp):
    user_pool = cognito_idp.create_user_pool(
        PoolName='test-user-pool',
    )
    user_pool_id = user_pool['UserPool']['Id']

    with mock.patch.dict(os.environ, {'COGNITO_USER_POOL_ID': user_pool_id}):
        yield user_pool

    cognito_idp.delete_user_pool(
        UserPoolId=user_pool_id
    )
```

Then, I created test with the above fixtures injected for the function that calls the line mentioned in the first code block. Line executes correctly and tests return correct results.

After deploying my code to AWS, the line however returns an error:
```
[ERROR] InvalidParameterException: An error occurred (InvalidParameterException) when calling the ListUsers operation: Error while parsing filter.
```

After looking it up in [AWS docs](https://docs.aws.amazon.com/cognito/latest/developerguide/how-to-manage-user-accounts.html#cognito-user-pools-searching-for-users-using-listusers-api) I learn that correct filter expression must have double quotas around interpolated value. For example:
```
{
    "AttributesToGet": [
        "given_name", "phone_number", "email"
    ],
    "Filter": "family_name = \"Reddy\"",
    "Limit": 10,
    "UserPoolId": "us-east-1_samplepool"
}
```
(notice `\"` around `Reddy`)

After changing my code to:
```
boto3.client('cognito-idp', aws_access_key_id='redacted', aws_secret_access_key='redacted').list_users(UserPoolId=os.getenv('COGNITO_USER_POOL_ID'), Filter='preferred_username= "nickname0"')
```
both tests with moto mocks and code deployed to AWS execute properly.

In order to fully and properly reflect behavior of AWS API, moto's `cognito-idp` client should throw the same error while parsing filter expression without double quotas around value.

Versions:
```
moto~=2.0.5
boto3~=1.17.49
botocore~=1.20.49
```

--- a/moto/cognitoidp/responses.py
+++ b/moto/cognitoidp/responses.py
@@ -335,7 +335,7 @@ class CognitoIdpResponse(BaseResponse):
             }
             comparisons = {"=": lambda x, y: x == y, "^=": lambda x, y: x.startswith(y)}
 
-            match = re.match(r"([\w:]+)\s*(=|\^=)\s*\"(.+)\"", filt)
+            match = re.match(r"([\w:]+)\s*(=|\^=)\s*\"(.*)\"", filt)
             if match:
                 name, op, value = match.groups()
             else:

--- a/tests/test_cognitoidp/test_cognitoidp.py
+++ b/tests/test_cognitoidp/test_cognitoidp.py
@@ -1233,6 +1233,10 @@ def test_list_users():
     result["Users"].should.have.length_of(1)
     result["Users"][0]["Username"].should.equal(user0_username)
 
+    # empty value Filter should also be supported
+    result = conn.list_users(UserPoolId=user_pool_id, Filter='family_name=""')
+    result["Users"].should.have.length_of(0)
+
 
 @mock_cognitoidp
 def test_list_users_incorrect_filter():

